### PR TITLE
TypeScript tooling fixes

### PR DESCRIPTION
**Pull request by brad-decker**
## Explanation
This introduces a few tiny changes to our existing tooling to work better with ts files.

1. tsconfig.json extended to ignore some of our build output directories, which cause typescript to slow down a fair bit and warn about excluding large folders.
2. jest.config.js extended to consider coverage of ts files, and to run .ts tests in more folders
3. verify-locale script extended to consider ts files in more locations.

## Manual Testing Steps
1. pull branch
2. `yarn`
3. run `yarn storybook` 
4. Ensure vscode isn't slow when in a .ts file (you can do this on a different branch to see what it looks like, or remove some of the tsconfig entries on this branch)

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by darkwing — DISMISSED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -18,10 +18,14 @@
...
     "**/*.test.tsx",
+    "**/jest-coverage/**/*",
+    "**/__mocks__/**/*",
+    "**/storybook-build/**/*",
     ".storybook/**/*",
     "builds/**/*",
     "dist/**/*",
-    "node_modules/**"
+    "node_modules/**",
+    "development/ts-migration-dashboard/**/*"
```
Comment (mcmire) [inline tsconfig.json:28]: There are TypeScript files inside of the dashboard code. Assuming that the goal is to exclude output files, should this be `development/ts-migration-dashboard/build/**/*` and `development/ts-migration-dashboard/intermediate/**/*`?

Comment (brad-decker) [reply tsconfig.json]: Done :)

File: development/verify-locale-strings.js [hunk 2]
```diff
@@ -180,6 +180,7 @@ async function verifyEnglishLocale() {
   const javascriptFiles = await glob(
     [
```
Comment (mcmire) [inline development/verify-locale-strings.js:181]: I wonder if this array is worth moving to a constant at the top. I guess this is fine for now though.

**Review by mcmire — APPROVED** [review]
Makes sense to me!

**Review by DDDDDanica — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**